### DISCUSSION TITLE
Getting error on Webview2

### DISCUSSION THREAD

**Discussion by tilankak** [MicrosoftEdge/WebView2Feedback#4051] [Q&A]
I am getting following error on my application. When I try initialize Webview2 in following code.

await webView2.EnsureCoreWebView2Async(null);

The error message is 

System.Runtime.InteropServices.COMException: 'The group or resource is not in the correct state to perform the requested operation. (Exception from HRESULT: 0x8007139F)'

My app details 
Winforms
.Net framework 4.7.2

webview2 version = 1.0.1774.30

**Comment by plantree** [comment]
Hi @tilankak,

Thanks for your feedback.

If you're available, could you provide a simple demo to help reproduce? Thanks!

**Comment by hatton** [comment]
I have this come and go, seemingly with each evergreen update of WV2 runtime. When things are broken, this code reproduces it

```

namespace WinFormsApp1
{
    public partial class Form1 : Form
    {
        WebView2 webView;
        public Form1()
        {
            InitializeComponent();

            webView = new Microsoft.Web.WebView2.WinForms.WebView2();
            webView.CreationProperties = new CoreWebView2CreationProperties();

            // this fixes it:
            //webView.CreationProperties.BrowserExecutableFolder = "C:\\Program Files (x86)\\Microsoft\\EdgeCore\\114.0.1823.37";

            webView.Dock = DockStyle.Fill;
            Controls.Add(webView);
        }

        private async void Form1_Load(object sender, EventArgs e)
        {
            await webView.EnsureCoreWebView2Async(null);
            webView.CoreWebView2.Navigate("https://www.bing.com");
        }
    }
}
```

related: https://github.com/MicrosoftEdge/WebView2Feedback/issues/3460

**Comment by plantree** [comment]
Hi @hatton,

Did you mean that the issue will happen during the update of runtime?

**Comment by hatton** [comment]
I mean that when it happens, I go to look at the runtime and find that it has been updated that day. Unfortunately, since Windows does not allow me to uninstall or roll back to a previous version, it is not possible to know the cause for sure.

**Comment by plantree** [comment]
Hi @hatton,

According to my understanding, your program can run normally, but there will be problems after the runtime update? And specify the `BrowserExecutableFolder` could fix?

**Comment by hatton** [comment]
Correct. Previously shipping version of our software stopped working in the middle of my workday. Went to check the version of WV2, found that a new version had been installed that very day. This is the second time this has happened. Last time, a new update eventually arrived and fixed the problem. This time, it has not happened yet. Meanwhile, setting BrowserExecutableFolder to an Edge folder fixes it.

Meanwhile, there seems to be no way to actually uninstall WV2 so I can do a full uninstall/reinstall.

**Comment by plantree** [comment]
Hi @hatton,

It seems that your program cannot work in a given version. Uninstall/reinstall might be a bit complicated, and you can try another way: package the given version runtime into the program. There is a related documentation here: https://learn.microsoft.com/en-us/microsoft-edge/webview2/concepts/distribution. 

Hope to solve your problem :) .

**Comment by hatton** [comment]
@plantree that is not the problem. The 7 other people on our team all use the same WebView2 version, with the same software.  Previously, we have been using the gecko engine, this is the first version where we are switching to WV2. Soon it will go beta and we'll find out how common my situation is. I hope it is very rare, but it's not logical to think that I will be the only one, with my new, clean system.

We're not bundling WV2 into our installer because our software is for people doing literacy in poor & developing countries, so we need to keep our installer size to a minimum.

> Uninstall/reinstall might be a bit complicated

Is it even possible? In #3460 I requested documentation for this.

**Comment by plantree** [comment]
Hi @hatton,

It seems that the issue is sporadic and hard to reproduce, and we can fix this case by case.

As for the install/unistall, thanks @novac42 for checking that.
